flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/resources/Resource.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.kubernetes.kubeclient.resources;
+
+import org.apache.flink.configuration.Configuration;
+
+/**
+ * Represent a kubernetes resource.
+ */
+public abstract class Resource<T> {
+
+	protected T internalResource;
```

Comment:
given we have getter for `internalResource` and `flinkConfig` they could be `private final`; just notice their setters are not into used at all.

Reply:
There is code

```java
subclass extends Resource {
  super(flinkConfig);
  this.internalResource = internalResource;
}
```

which could be 

```java
  super(flinkConfig, internalResource);
```

Reply:
Nice suggestion. I will update the `Resource` and implementation class.